src/cript/api/api.py

```diff
@@ -429,6 +470,126 @@ def save(self, project: Project) -> None:
         if response["code"] != 200:
             raise CRIPTAPISaveError(api_host_domain=self._host, http_code=response["code"], api_response=response["error"])
 
+    def upload_file(self, file_path: Union[Path, str]) -> str:
```

Comment:
not sure this should be a public function.
Typically users should see or use this directly.

Reply:
I was not fully sure either, but I thought that it made sense to me to have `upload_file` to be public so that if the user wants to upload it themselves within their script they can

and within the file node we check that if the file source starts with `http` then we accept it, otherwise it would be a local file path and we upload that to our AWS S3, get a link, and put the link into the file source